---
title: "deseqOculina_symbiont"
author: "Hannah Aichelman"
date: "10/22/2019"
output: 
  html_document:
    toc: true
    theme: united
---


### Install and libray packages
```{r InstallPackages, eval=FALSE, message=FALSE, warning=FALSE, include=FALSE}
# This code is from Colleen Bove, lets us library the old version of DESeq (to do array quality metrics) even running versions of R after 3.6

## get Bioconductor packages
if (!requireNamespace("BiocManager"))
install.packages("BiocManager")
BiocManager::install("DESeq2")

## installing WGCNA:
#source("http://bioconductor.org/biocLite.R")
BiocManager::install(c("AnnotationDbi", "impute", "GO.db", "preprocessCore"))
BiocManager::install("WGCNA", dependencies=TRUE)
BiocManager::install("arrayQualityMetrics", dependencies=TRUE) # use this arrayQualityMetrics install if using later versions of R (3.6.3 works)
BiocManager::install("affycoretools", dependencies=TRUE)
#repos="http://cran.us.r-project.org"
## R version 3.6 is funky with arrayQualityMetrics and DESeq so need this work around:
install.packages("ape", lib = "/Library/Frameworks/R.framework/Versions/4.0/Resources/library")
library(ape, lib.loc = "/Library/Frameworks/R.framework/Versions/4.0/Resources/library")
install.packages("magick", lib = "/Library/Frameworks/R.framework/Versions/4.0/Resources/library", dependencies = FALSE)
library(magick, lib.loc = "/Library/Frameworks/R.framework/Versions/4.0/Resources/library")
BiocManager::install("arrayQualityMetrics", type = "source", checkBuilt = TRUE, lib.loc = "/Library/Frameworks/R.framework/Versions/4.0/Resources/library")
BiocManager::install("DESeq", type = "source", checkBuilt = TRUE, lib.loc = "/Library/Frameworks/R.framework/Versions/4.0/Resources/library")
```

```{r Libraries, message=FALSE, warning=FALSE}
library(tidyr)
library(plyr)
library(dplyr)
library(DESeq2)
library(ggplot2)
library(affycoretools)
library(arrayQualityMetrics)
library(genefilter)
#library(DESeq)
library(cowplot)
library(readr)
library(RColorBrewer)
library(gplots)
library(knitr)
library(plotly)
library(vegan)
library(kableExtra)
library(reshape2)
library(prettydoc)
library(VennDiagram)
```

### Counts file and experimental design

Read in Breviolum psygmophilum symbiont in culture counts file
```{r,cache=TRUE, echo=FALSE}
countDataCulture <- read.table("/Users/hannahaichelman/Documents/BU/Host_Buffering/MPCC_2018/Sym_analyses/CountsFiles/B_psygmophilum_counts.txt")
```


Set up experimental design matrix
```{r,cache=TRUE}
# rename the annoying name and add isogroup to the row name
names(countDataCulture)=c( "Control_1.1", "Control_1", "Control_2.1", "Control_2", "Control_3.1", "Control_3", "Control_4.1", "Control_4", "Cool_1", "Cool_2", "Cool_3", "Cool_4","Heat_1.1", "Heat_1", "Heat_2.1", "Heat_2", "Heat_3.1", "Heat_3", "Heat_4.1", "Heat_4")
row.names(countDataCulture)=sub("", "isogroup", rownames(countDataCulture))
head(countDataCulture)

treatment = as.factor(sapply(strsplit(colnames(countDataCulture), split = "_"), "[[", 1)) %>%
  revalue(c("Control" = "control", "Cool" = "cold", "Heat" = "heat"))

expDesign = data.frame(treatment) 
expDesign$sample = colnames(countDataCulture)
```

Experimental Design key with sample names 
```{r,cache=TRUE, echo=FALSE}
kable(expDesign) %>%
  kable_styling(bootstrap_options = "striped", full_width = F)
```
Descriptive Summary stats of the mapped reads. 

```{r echo=FALSE}
stats.per.sample = data.frame(t(do.call(cbind, lapply(countDataCulture, summary))))
      stats.per.sample$libsum = apply(countDataCulture, 2, sum) ## libsum
      stats.per.sample$perc05 = apply(countDataCulture, 2, quantile, 0.05)
      stats.per.sample$perc10 = apply(countDataCulture, 2, quantile, 0.10)
      stats.per.sample$perc90 = apply(countDataCulture, 2, quantile, 0.90)
      stats.per.sample$perc95 = apply(countDataCulture, 2, quantile, 0.95)
      stats.per.sample$zeros = apply(countDataCulture==0, 2, sum)
      stats.per.sample$percent.zeros = 100*stats.per.sample$zeros/nrow(countDataCulture)
write.csv(stats.per.sample, file = "/Users/hannahaichelman/Documents/BU/Host_Buffering/MPCC_2018/Sym_analyses/tables/sym_culture_summary_table.csv", quote = FALSE)
      
kable(stats.per.sample) %>%
  kable_styling(bootstrap_options = "striped", full_width = F)
```

#Outlier Detection
Conduct array quality metrics to identify outliers  
Skip this when running - output from array quality metrics provided in Github

```{r, cache=TRUE}
real=newCountDataSet(countDataCulture,expDesign) 
real=estimateSizeFactors(real)
```

```{r, fig.align='center', cache=TRUE }
plot(sort(sizeFactors(real)), main = "Ocuina Symbiont Size Factors") 
```
<br>
Size factors of each sample used to normalize between libraries. 
<br> 
<br> 
Create a directory with a bunch of output figures from arrayQualityMetrics() for you to explore if their are outliers. 

```{r message=FALSE, warning=FALSE, eval = FALSE}
cds=estimateDispersions(real,method="blind")
vsdBlind=varianceStabilizingTransformation(cds)
arrayQualityMetrics(vsdBlind,intgroup=c("treatment"), force=TRUE, outdir = "Oculina_brown_sym_arrayqualitymetrics") # this makes a directory "arrayQualityMetrics" and outputs various reports on outliers
```

Look at total counts across Oculina host and symbiont samples
```{r}
totalCountshost=colSums(countDataHost)
totalCountshost
barplot(totalCountshost, col=treatment, ylab="raw counts", main = "Oculina host total counts")

min(totalCountshost) 
max(totalCountshost) 


totalCountssym=colSums(countDataSym)
totalCountssym
barplot(totalCountssym, col=treatment, ylab="raw counts", main = "Oculina symbiont total counts")

min(totalCountssym) 
max(totalCountssym) 

totalCountsculture=colSums(countDataCulture)
totalCountsculture
barplot(totalCountsculture, col=treatment, ylab="raw counts", main = "B. psygmophilum culture total counts")

min(totalCountsculture) 
max(totalCountsculture) 

```
####Outlier Conclusisons

Do not need to remove any outliers for the Oculina host or symbiont. This is considering brown/symbiotic individuals only though.

Skipped this step for the Culture because I am now running R v4.0.1 and I cannot library the original DESeq package. 

***

#Differential Expression

***

```{r warning=FALSE, cache=TRUE}
dds<-DESeqDataSetFromMatrix(countData=countDataCulture, colData=expDesign, design=~treatment) #can only test for the main effects of treatment
dds = DESeq(dds)
results = results(dds)
summary(results)
```



Let's check to see if we set up our contrast correctly. We should have the treatment condition first and the control second in the log2 fold change (MLE) output. 

```{r, cache=TRUE}
head(results)
```

```{r echo=FALSE, cache=TRUE}
norm.counts.culture = counts(dds, normalized = TRUE) # these are the counts DESeq uses
write.csv(norm.counts.culture,"/Users/hannahaichelman/Documents/BU/Host_Buffering/MPCC_2018/Sym_analyses/tables/normalized_counts_culture.csv") #these are all counts, not considering treatment comparisons
              
norm.counts.stats = data.frame(
  min = apply(norm.counts.culture, 2, min),
  mean = apply(norm.counts.culture, 2, mean),
  median = apply(norm.counts.culture, 2, median), 
  max = apply(norm.counts.culture, 2, max),
  zeros = apply(norm.counts.culture == 0, 2, sum), 
  percent.zeros = 100* apply(norm.counts.culture == 0, 2, sum) / nrow(norm.counts.culture) 
)

kable(norm.counts.stats) %>%
  kable_styling(bootstrap_options = "striped", full_width = F)
```

Lets do a rlogged transformation, which is useful various unsupervised clustering analyses. Be sure the include blind = TRUE as it doesn't normalize the data with any priors from our experimental design. 

```{r, cache=TRUE}
rlogged = rlogTransformation(dds, blind = TRUE)
```

Hot versus control expression comparison
```{r}
##second term is the "control"
res_hot_culture <- results(dds, contrast=c("treatment","heat","control"))
head(res_hot_culture)
#how many FDR < 10%?
table(res_hot_culture$padj<0.05)
# 0.1=5415
# 0.05=4330
# 0.01=2860
summary(res_hot_culture)
write.table(res_hot_culture, file="/Users/hannahaichelman/Documents/BU/Host_Buffering/MPCC_2018/Sym_analyses/tables/culture_hot_results.txt", quote=F, sep="\t")
```

MA plot, red dots indicate genes significantly differentially expressed at FDR < 0.1 

```{r, cache=TRUE, fig.align='center'}
DESeq2::plotMA(res_hot_culture, main = "Heat vs Control, Culture")
```


Cold versus control expression comparison
```{r}
##second term is the "control"
res_cold_culture <- results(dds, contrast=c("treatment","cold","control"))
head(res_cold_culture)
#how many FDR < 10%?
table(res_cold_culture$padj<0.1)
# 0.1=7163
# 0.05=5834
# 0.01=4057
summary(res_cold_culture)
write.table(res_cold_culture, file="/Users/hannahaichelman/Documents/BU/Host_Buffering/MPCC_2018/Sym_analyses/tables/culture_cold_results.txt", quote=F, sep="\t")
```

MA plot, red dots indicate genes significantly differentially expressed at FDR < 0.1 

```{r, cache=TRUE, fig.align='center'}
DESeq2::plotMA(res_cold_culture, main = "Cold vs Control, Culture")
```


## Get RLDPVAL table
```{r, cache=TRUE, fig.align='center'}
#get pvals
valcold=cbind(res_cold_culture$pvalue, res_cold_culture$padj)
head(valcold)
colnames(valcold)=c("pval.cold.culture", "padj.cold.culture")
length(valcold[,1])
table(complete.cases(valcold))

valhot=cbind(res_hot_culture$pvalue, res_hot_culture$padj)
head(valhot)
colnames(valhot)=c("pval.hot.culture", "padj.hot.culture")
length(valhot[,1])
table(complete.cases(valhot))

#make rlog data and pvals table
rld=assay(rlogged)
head(rld)
length(rld[,1])

rldpvals=cbind(rld,valcold, valhot)
head(rldpvals)
dim(rldpvals)
# [1] 28265    24
table(complete.cases(rldpvals))
# FALSE  TRUE 
# 8768 19497

write.csv(rldpvals, "/Users/hannahaichelman/Documents/BU/Host_Buffering/MPCC_2018/Sym_analyses/tables/Culture_RLDandPVALS.csv", quote=F)

```


## Make a barplot of differentially expressed genes
```{r, cache=TRUE, fig.align='center'}
summary(res_hot_culture)
# up = 2852
# down = 2563
summary(res_cold_culture)
# up = 3963
# down = 3200

culture_degs = data.frame(
  temperature = factor(c("Heat", "Heat", "Cold", "Cold")),
  type = factor(c("Sym_InCulture", "Sym_InCulture", "Sym_InCulture", "Sym_InCulture")),
  direction = factor(c("up","down","up","down")),
  temperature_direction = factor(c("heat_up","heat_down", "cold_up","cold_down")),
  number = c(2852, -2563, 3963, -3200)
)

annotate_text = data.frame(
  label = c("2852", "2563", "3963", "3200"),
  temperature = factor(c("Heat", "Heat", "Cold", "Cold")),
  type = factor(c("Sym_InCulture", "Sym_InCulture", "Sym_InCulture", "Sym_InCulture")),
  direction = factor(c("up","down","up","down")),
  temperature_direction = factor(c("heat_up","heat_down", "cold_up","cold_down")),
  x = c(1,1,2,2),
  y = c(4200,-3450,3100,-2900)
)

cols = c("cold_up" = "#74c476", "cold_down" =  "#c7e9c0","heat_up" = "#fd8d3c", "heat_down" ="#fdd0a2")

deg_barplot = ggplot(culture_degs, aes(x = temperature, y = number, fill=temperature_direction)) +
  geom_bar(stat = "identity", position = position_stack(), color = "black") +
  scale_fill_manual(values = cols) +
  xlab("Temperature Treatment") +
  ylab("Differentially expressed genes") +
  theme_bw() +
  theme(legend.position = "none") +
  facet_grid(. ~ type) +
  #theme(panel.spacing = unit(0, "lines")) +
  geom_text(data = annotate_text, aes(x = x, y = y, label = label))
deg_barplot

ggsave(deg_barplot, file = "/Users/hannahaichelman/Documents/BU/Host_Buffering/MPCC_2018/Sym_analyses/plots/culture_deg_barplot.pdf", width=5, height=4, units=c("in"), useDingbats=FALSE)
```

##Principal Component Analyses 

<br>
First we create a PCA data frame and calculate the variance estimated by PC1 and PC2
```{r cache=TRUE, fig.align="center"}
pcadata = DESeq2::plotPCA(rlogged, intgroup = c( "treatment"), returnData = TRUE)
percentVar = round(100 * attr(pcadata, "percentVar"))
pca = prcomp(t(assay(rlogged)), center = TRUE, scale. = FALSE)
```
<br><br>

Using adonis from library(vegan) we can see if there are any significant differences based on treatment (as is illustrated in the model displayed below) 
```{r cache=TRUE}
adonis2(pca$x ~ treatment, data = pcadata, method = 'eu')

#           Df SumOfSqs     R2      F Pr(>F)    
# treatment  2   9958.4 0.5414 10.035  0.001 ***
# Residual  17   8435.4 0.4586                  
# Total     19  18393.8 1.0000                  
```
<br><br>
Significant effect of treatment (p = 0.001)
```{r cache=TRUE, fig.align="center"}
# symbionts in culture, so using open circles but same colors as in symbiosis
cols_sym = c("control" = "#a6611a", "cold" = "#74c476", "heat" = "#fd8d3c")

pca_culture = DESeq2::plotPCA(rlogged, returnData = TRUE, intgroup = c("treatment") ) %>% 
      ggplot(aes(x = PC1, y = PC2)) +
      geom_point(aes(colour = treatment), size = 3, shape = 21, stroke = 1) +
      scale_colour_manual(values = cols_sym,
                          name = "Treatment") +
      stat_ellipse(aes(color=rlogged$treatment), type = "t", linetype = 2, lwd = 1) +
      xlab(paste0("PC1: ",percentVar[1],"% variance")) +
      ylab(paste0("PC2: ",percentVar[2],"% variance")) +
      #labs(title = "~ treatment, p<0.001")+
      theme_bw()
pca_culture

ggsave(pca_culture, file = "/Users/hannahaichelman/Documents/BU/Host_Buffering/MPCC_2018/Sym_analyses/plots/culture_pca.pdf", width=5, height=4, units=c("in"), useDingbats=FALSE)
```
Principal component analyses highlighting treatment groups. Treatment has a significant affect (p < 0.001) on overall expression

<br><br>

** Should try to do this for culture individual flask at some point. 
```{r cache=TRUE, fig.align="center"}
DESeq2::plotPCA(rlogged, returnData = TRUE, intgroup = c("treatment", "genotype") ) %>% 
      ggplot(aes(x = PC1, y = PC2)) +
      geom_point(aes(colour = genotype), size = 3) +
      xlab(paste0("PC1: ",percentVar[1],"% variance")) +
      ylab(paste0("PC2: ",percentVar[2],"% variance")) +
      labs(title = "~ genotype, p=0.2")+
      theme_cowplot()
```
Principal component analyses highlighting genotypes of coral. Genotype does not have a significant affect (p < 0.2) on overall expression

### PCA Plasticity 

PCA plasticity functions written by C. Bove.

This function allows you to save all principal components 
```{r}
plotPCA_allPCs <- function (object, intgroup = "condition", ntop = 500, returnData = FALSE)
{
  rv <- rowVars(assay(object)) # Variance estimates for each row (column) in a matrix.
  select <- order(rv, decreasing = TRUE)[seq_len(min(ntop, length(rv)))] # orders genes from largest -> smallest and pulls ntop (default 500) genes
  pca <- prcomp(t(assay(object)[select, ]), center = TRUE, scale. = FALSE) # performs PCA on ntop (default 500) from dataset
  percentVar <- pca$sdev^2/sum(pca$sdev^2)
  if (!all(intgroup %in% names(colData(object)))) {
    stop("the argument 'intgroup' should specify columns of colData(dds)")
  }
  intgroup.df <- as.data.frame(colData(object)[, intgroup, drop = FALSE])
  group <- if (length(intgroup) > 1) {
    factor(apply(intgroup.df, 1, paste, collapse = ":"))
  }
  else {
    colData(object)[[intgroup]]
  }
  d <- data.frame(group = group, intgroup.df, name = colnames(object), pca$x)
  if (returnData) {
    attr(d, "percentVar") <- percentVar[1:2]
    return(d)
  }
  ggplot(data = d, aes_string(x = "PC1", y = "PC2", color = "group")) +
    geom_point(size = 3) + xlab(paste0("PC1: ", round(percentVar[1] *
                                                        100), "% variance")) + ylab(paste0("PC2: ", round(percentVar[2] *
                                                                                                            100), "% variance")) + coord_fixed()
}
```

```{r eval=FALSE, include=FALSE}
# source the script Colleen wrote with the different functions to calculate plasticity
source("/Users/hannahaichelman/Documents/BU/Host_Buffering/MPCC_2018/Sym_analyses/GE_Plasticity/PlasticityCustomFunctions_means.R")

# use modified plotPCA function to give you all PCs
pcadata.plast = plotPCA_allPCs(rlogged, intgroup = c("treatment"), returnData = TRUE)


pca_plast <-  PCAplast(pca = pcadata.plast[,c(4:ncol(pcadata.plast))], # PCA dataframe containing the PCA eigenvalues
                data = pcadata.plast[,c(1:3)], # condition/treatment data 
                sample_ID = "name", # name of column with unique ID per sample (if blank, will pull rownames for this)
                num_pca = "2", # number of PCs to include in measure (default is 'all' if left blank)
                control_col = "treatment", # name of 'treatment' column
                control_lvl = "control") # control level of the treatment 
```

Now plot PCA plasticity, separately for host and symbiont
```{r}
library(Rmisc)
library(performance)

cols_sym = c("control" = "#a6611a", "cold" = "#74c476", "heat" = "#fd8d3c")

dist_summ = summarySE(data = pca_plast, measurevar = "dist", groupvars = "treatment")

# Stats
lm = lm(dist ~ treatment, data = pca_plast)
summary(lm)
# Coefficients:
#               Estimate Std. Error t value Pr(>|t|)    
# (Intercept)     0.3660     0.2196   1.666    0.114    
# treatmentcold  16.9571     0.3804  44.576   <2e-16 ***
# treatmentheat  15.6321     0.3106  50.328   <2e-16 ***

aov = aov(dist ~ treatment, data = pca_plast)
check_model(aov) # check assumptions visually

summary(aov)
#             Df Sum Sq Mean Sq F value Pr(>F)    
# treatment    2 1244.8   622.4    1613 <2e-16 ***
# Residuals   17    6.6     0.4                   

TukeyHSD(aov)
# $treatment
#                   diff       lwr        upr     p adj
# cold-control 16.957109 15.981219 17.9329993 0.0000000
# heat-control 15.632067 14.835256 16.4288776 0.0000000
# heat-cold    -1.325043 -2.300933 -0.3491524 0.0075959

# Plot symbiont in host GE plasticity
plast_plot = ggplot(dist_summ, aes(x = treatment, y = dist)) +
      geom_point(aes(colour = treatment), size = 4) +
      geom_errorbar(aes(x = treatment, ymin = dist - sd, ymax = dist + sd, colour = treatment), width = 0.2) +
      scale_color_manual(values = cols_sym) +
      scale_fill_manual(values = cols_sym) +
      xlab("Treatment") +
      ylab("Plasticity") +
      labs(title = "Gene Expression Plasticity") +
      theme_cowplot() +
      theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1), legend.position = "none")
plast_plot

ggsave(plast_plot, filename = "/Users/hannahaichelman/Documents/BU/Host_Buffering/MPCC_2018/Sym_analyses/plots/PCA_plasticity_2PCs_syminculture.pdf", width=6, height=5, units=c("in"), useDingbats=FALSE)

# boxplot with points overlaid
plast_nocontrol = pca_plast %>%
  dplyr::filter(treatment != "control")

plast_boxplot <- ggplot(plast_nocontrol, aes(x = treatment, y = dist)) +
  geom_boxplot(outlier.shape = NA,
               alpha = 0.8,
               aes(fill = treatment))+
  scale_fill_manual(values = cols_sym) + # for boxplot
    geom_jitter(size = 2,
              position=position_jitter(0.2), 
              alpha=0.99,
              color = 'black',
              shape = 21) +
  scale_color_manual(values = cols_sym) + # for jittered points
  ylab("Gene Expression Plasticity") +
  theme_bw() +
  #theme(legend.position = "none") +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1, face ="bold"), legend.position = "none") +
  theme(axis.title.x = element_blank())
plast_boxplot

ggsave(plast_boxplot, filename = "/Users/hannahaichelman/Documents/BU/Host_Buffering/MPCC_2018/Sym_analyses/plots/PCA_plasticity_boxplot_syminculture.pdf", width=2.5, height=4, units=c("in"), useDingbats=FALSE)

```


##Heatmaps
***
<br> <br> 

Overall expression, sample by distance. 

```{r, cache=TRUE, fig.align="center"}
sampleDists <- as.matrix(dist(t(assay(rlogged))))
heatmap.2(as.matrix(sampleDists), key=F, trace="none",
          col=colorpanel(100, "black", "white"),
          margin=c(10, 10))
```

<br><br>
<b>Individual genes heatmap </b> 
<br> 
Now plotting the z scores, as heatmap2 creates nice clean clusters by doing this. Upregulation indicated by warmer colors, downregulation indicated by cooler colors.

First for heat vs. control comparison
```{r, cache=TRUE, fig.align='center'}
norm_counts = read.csv("/Users/hannahaichelman/Documents/BU/Host_Buffering/MPCC_2018/Sym_analyses/tables/normalized_counts_culture.csv")
hm = read.table("/Users/hannahaichelman/Documents/BU/Host_Buffering/MPCC_2018/Sym_analyses/tables/culture_hot_results.txt", header=TRUE, col.names=c("X", "baseMean", "log2FoldChange", "lfcSE", "stat", "pvalue","padj")) %>% as_tibble() %>%
  filter(padj < 0.1) %>% # only want the most DEGs
  select(X) %>%
  merge(norm_counts, by.x = "X", by.y = "X")  # turn into a countdatafile
  row.names(hm) = hm$X
  hm$X = NULL

## Turning into z-score table
hm.z = data.matrix(hm)
hm.z = sweep(hm.z, 1L, rowMeans(hm.z), check.margin = FALSE)
hm.z.sx = apply(hm.z, 1L, sd)
hm.z = sweep(hm.z, 1L, hm.z.sx, "/", check.margin = FALSE)
hm.z = data.matrix(hm.z)

colour = colorRampPalette(rev(c("chocolate1","#FEE090","grey10", "cyan3","cyan")))(100)
heatmap.2(hm.z, col = colour, Rowv = TRUE, Colv = TRUE, scale = "row", 
          dendrogram = "both",
          trace = "none", 
          margin = c(5,15))
```

Now for cold vs. control comparison
```{r, cache=TRUE, fig.align='center'}
norm_counts = read.csv("/Users/hannahaichelman/Documents/BU/Host_Buffering/MPCC_2018/Sym_analyses/tables/normalized_counts_culture.csv")
hm = read.table("/Users/hannahaichelman/Documents/BU/Host_Buffering/MPCC_2018/Sym_analyses/tables/culture_cold_results.txt", header=TRUE, col.names=c("X", "baseMean", "log2FoldChange", "lfcSE", "stat", "pvalue","padj")) %>% as_tibble() %>%
  filter(padj < 0.1) %>% # only want the most DEGs
  select(X) %>%
  merge(norm_counts, by.x = "X", by.y = "X")  # turn into a countdatafile
  row.names(hm) = hm$X
  hm$X = NULL

## Turning into z-score table
hm.z = data.matrix(hm)
hm.z = sweep(hm.z, 1L, rowMeans(hm.z), check.margin = FALSE)
hm.z.sx = apply(hm.z, 1L, sd)
hm.z = sweep(hm.z, 1L, hm.z.sx, "/", check.margin = FALSE)
hm.z = data.matrix(hm.z)

colour = colorRampPalette(rev(c("chocolate1","#FEE090","grey10", "cyan3","cyan")))(100)
heatmap.2(hm.z, col = colour, Rowv = TRUE, Colv = TRUE, scale = "row", 
          dendrogram = "both",
          trace = "none", 
          margin = c(5,15))
```

##Venn Diagrams

Read in our result tables and isolate gene lists that are significant in each experiment. 

```{r cache=TRUE}
hot = read.table("/Users/hannahaichelman/Documents/BU/Host_Buffering/MPCC_2018/Sym_analyses/tables/culture_hot_results.txt", header = TRUE)
hot = row.names(hot[hot$padj<0.05 & !is.na(hot$padj),])

cold = read.table("/Users/hannahaichelman/Documents/BU/Host_Buffering/MPCC_2018/Sym_analyses/tables/culture_cold_results.txt", header = TRUE)
cold = row.names(cold[cold$padj<0.05 & !is.na(cold$padj),])
```
<br><br>

Set up a venn diagram looking for similar genes in each experiment. 
```{r cache=TRUE}
all_shared = list("Hot" = hot, "Cold" = cold)
prettyvenn=venn.diagram(
      x = all_shared,
      filename=NULL,
      #col = "transparent",
      fill = c("#fd8d3c", "#74c476"),
      alpha = 0.5,
      # label.col = c("darkred", "white", "darkgreen", "white", "white", "white", "blue4"),
      cex = 2.5,
      fontfamily = "sans",
      fontface = "bold",
      cat.default.pos = "text",
      cat.col = "black",
      cat.cex = 2.5,
      cat.fontfamily = "sans",
      cat.dist = c(0.08, 0.08),
      cat.pos = 1
    );
grid.draw(prettyvenn)
```



##Gene Ontology
Make Gene Ontology input table for MWU for hot treatment
```{r message=FALSE, warning=FALSE, eval = FALSE}
culture_hot_res = read.table("/Users/hannahaichelman/Documents/BU/Host_Buffering/MPCC_2018/Sym_analyses/tables/culture_hot_results.txt")
head(culture_hot_res)
culture_hot_res$isogroup=row.names(culture_hot_res)

go_input_hot_culture = culture_hot_res %>%
  mutate(mutated_p = -log(pvalue)) %>%
  mutate(mutated_p_updown = ifelse(log2FoldChange < 0, mutated_p*-1, mutated_p*1)) %>%
  select(isogroup, mutated_p_updown) %>%
  na.omit()

nrow(go_input_hot_culture)
head(go_input_hot_culture)
colnames(go_input_hot_culture) <- c("gene", "pval")
head(go_input_hot_culture)
write.csv(go_input_hot_culture, file="/Users/hannahaichelman/Documents/BU/Host_Buffering/MPCC_2018/Sym_analyses/GO_Analyses/culture_hot_GO.csv", quote=F, row.names=FALSE)

```

Now for cold treatment
```{r message=FALSE, warning=FALSE, eval = FALSE}
culture_cold_res = read.table("/Users/hannahaichelman/Documents/BU/Host_Buffering/MPCC_2018/Sym_analyses/tables/culture_cold_results.txt")
head(culture_cold_res)
culture_cold_res$isogroup=row.names(culture_cold_res)

go_input_cold_culture = culture_cold_res %>%
  mutate(mutated_p = -log(pvalue)) %>%
  mutate(mutated_p_updown = ifelse(log2FoldChange < 0, mutated_p*-1, mutated_p*1)) %>%
  select(isogroup, mutated_p_updown) %>%
  na.omit()

nrow(go_input_cold_culture)
head(go_input_cold_culture)
colnames(go_input_cold_culture) <- c("gene", "pval")
head(go_input_cold_culture)
write.csv(go_input_cold_culture, file="/Users/hannahaichelman/Documents/BU/Host_Buffering/MPCC_2018/Sym_analyses/GO_Analyses/culture_cold_GO.csv", quote=F, row.names=FALSE)
```


